rebasehelper/application.py

```diff
@@ -194,6 +195,9 @@ def _prepare_spec_objects(self):
                 raise RebaseHelperError('Could not determine latest upstream version '
                                         'and no SOURCES argument specified!')
 
+        if parse_version(self.conf.sources) <= parse_version(self.spec_file.get_version()):
```

Comment:
Beware that `self.conf.sources` doesn't have to be a version string, it can be filename of an source archive. See `SpecFile.set_version_using_archive()` below.